JavadocStyleGuide.md

```diff
@@ -0,0 +1,242 @@
+
+Custom Gradle tags:
+
+| # | Tag         | Usage                                        | Notes                                                                                                                                                                                                               |
+|---|-------------|----------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 1 | `@apiSpec`  | Adds a “API Requirements” heading            | A description that applies equally to all valid implementations of the method, including preconditions, postconditions, etc.                                                                                        |
+| 2 | `@apiNote`  | Adds a “API Note” heading                    | A commentary, rationale, or example pertaining to the API.                                                                                                                                                          |
+| 3 | `@implSpec` | Adds a “Implementation Requirements” heading | This is where the default implementation (or an overrideable implementation in a class) is specified.                                                                                                               |
+| 4 | `@implNote` | Adds a “Implementation Note” heading         | This section contains informative notes about the implementation, such as advice to implementors, or performance characteristics that are specific to the implementation in this class of this version of the JDK.  |
+
+When a block tag doesn't fit on a single line, continuation lines are indented four (or more) spaces from the position of the `@`.
+
+## 1.2 The summary fragment
+
+Each Javadoc block begins with a brief summary fragment - this is the first sentence up until the character `.` is encountered.
+
+This fragment is very important: it is the only part of the text that appears in certain contexts such as class and method indexes:
+
+```java
+/**
+ * Returns an Image object that can then be painted on the screen.
+ */
+```
+
+## 1.3 Code blocks and snippets
+
+`<pre>` is the default HTML tag for preformatted text.
+All code blocks and snippets should be wrapped in `<pre>...</pre>` at minimum.
+
+Code blocks are formatted using [`highlight.js`](https://highlightjs.org/) using the `<code>` tag.
+All code blocks and snippets should be wrapped in `<code class="language-*****"></code>` to be automatically highlighted.
+Make sure the correct language is added to the class name: `language-kotlin`, `language-groovy`, or `language-java`.
+
+A formatted and highlighted code snippet looks should look as follows:
+
+```java
+/**
+ * <pre class='groovy autoTested'><code class="language-kotlin">
```

Comment:
This should note the mechanism for automatic testing.  It's not enough to just add the `autoTested` class, your project also needs a test class that extends `AbstractAutoTestedSamplesTest` to run them.  Hopefully we'll get rid of this requirement soon.

There are also other "tag suffixes" that do different things, as visible here: https://github.com/gradle/gradle/blob/master/testing/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AbstractAutoTestedSamplesTest.groovy#L39-L48